tests/Agency/StoreTestAPI.cpp

```diff
@@ -0,0 +1,973 @@
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Ignacio Rodriguez
+/// @author Max Neunhoeffer
+/// @author Copyright 2021, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include <algorithm>
+#include "gtest/gtest.h"
+
+#include "Agency/Store.h"
+#include "Mocks/Servers.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/Compare.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+namespace arangodb {
+namespace tests {
+namespace store_test_api {
+
+class StoreTestAPI : public ::testing::Test {
+ public:
+  StoreTestAPI() : _server(), _store(_server.server(), nullptr) {
+  }
+ protected:  
+  arangodb::tests::mocks::MockCoordinator _server;
+  arangodb::consensus::Store _store;
+
+  auto readAndCheck(std::string const &json) 
+  {
+    try {
+      auto q {VPackParser::fromJson(json)};
+      auto result {std::make_shared<VPackBuilder>()};
+      _store.read(q, result);
+      return result;
+    }
+    catch(std::exception &ex) {
+      throw std::runtime_error(std::string(ex.what()) + " while trying to read " + json);
+    }
+  }
+
+  auto write(std::string const &json)
+  {
+    try {
+      auto q {VPackParser::fromJson(json)};
+      return _store.applyTransactions(q);
+    }
+    catch(std::exception &err) {
+      throw std::runtime_error(std::string(err.what()) + " while parsing " + json );
+    }
+  }
+
+  template<typename TOstream, typename TValueContainer>
+  static TOstream& insert_value_array(TOstream &out, TValueContainer const &src) {
+    out << "[";
+    if (!src.empty()) {
+      std::copy(src.begin(), src.end()-1, std::ostream_iterator<std::string>(out, ", "));
+      out << src.back();
+    }
+    out << "]";
+    return out;
+  }
+
+  auto write(std::vector<std::vector<std::string>> const &operations) {
+    std::stringstream ss;
+    ss << "[";
+    if (!operations.empty()) {
+      insert_value_array(ss, operations.front());
+      for (auto oi {operations.begin() + 1}; oi != operations.end(); ++oi) {
+        insert_value_array(ss, *oi);
+      }
+    }
+    ss << "]";
+    return write(ss.str());
+  }
+
+  auto transactAndCheck(std::string const &json) {
+    auto q {VPackParser::fromJson(json)};
+    auto results { _store.applyTransactions(q)};
+    return results;
+  }
+
+  auto writeAndCheck(std::string const &json)
+  {
+    try {
+      auto r {write(json)};
+      auto applied_all = std::all_of(r.begin(), r.end(), [](auto const &result){ return result == consensus::apply_ret_t::APPLIED; });
+      if (!applied_all)
```

Comment:
If you have a test, the `ASSERT_EQ` is enough, so you can remove the `if`. Besides, throwing here will be caught be the `catch` below which does not really seem to be sensible.

Reply:
Fine, but then the `ASSERT_TRUE` is unnecessary, since the code will never be reached if `applied_all` is `false`.